### ISSUE TITLE
matterbridge never speak in xmpp muc

### ISSUE THREAD

**Issue by jadjay** [42wim/matterbridge#339]
### Which version of matterbridge are you using?
1.6.3

### Please describe the expected behavior.
When design a gateway between xmpp and mattermost (inout for both), I expect that both xmpp->mm and mm->xmpp works
xmppMUC to mm works perfectly but 
mm to xmppMUC doesn't work at all, and I see messages in matterbridge log  

#### Use logs from running ```matterbridge -debug``` if possible.
Typing a message in MM channel
```
DEBU[2018-01-13T00:55:01+01:00] WsReceiver event: 
	model.WebSocketEvent{Event:"typing", Data:map[string]interface {}{"parent_id":"", "user_id":"mrjzniayh3fo9jut78uifrwoha"}, Broadcast:(*model.WebsocketBroadcast)(0xc420886700), Sequence:2}  module=matterclient

DEBU[2018-01-13T00:55:01+01:00] WsReceiver event: 
	model.WebSocketEvent{Event:"posted", Data:map[string]interface {}{"channel_display_name":"Centre-ville", "channel_name":"town-square", "channel_type":"O", "post":"{\"id\":\"6uho5jspibdnbmf5nq84dnwcnc\",\"create_at\":1515801301774,\"update_at\":1515801301774,\"edit_at\":0,\"delete_at\":0,\"is_pinned\":false,\"user_id\":\"mrjzniayh3fo9jut78uifrwoha\",\"channel_id\":\"etirmhnuctd7xgefxri6b6dg9a\",\"root_id\":\"\",\"parent_id\":\"\",\"original_id\":\"\",
		\"message\":\"mm vers xmpp\",
\"type\":\"\",\"props\":{},\"hashtags\":\"\",\"pending_post_id\":\"mrjzniayh3fo9jut78uifrwoha:1515801301701\"}", "sender_name":"SENDERNAME", 
"team_id":"trrjcgmydbd6zyfin15wred86h"}, Broadcast:(*model.WebsocketBroadcast)(0xc420886740), Sequence:3}  module=matterclient

DEBU[2018-01-13T00:55:01+01:00] map[string]interface {}
{"channel_display_name":"Centre-ville", "channel_name":"town-square", "channel_type":"O", "post":"{\"id\":\"6uho5jspibdnbmf5nq84dnwcnc\",\"create_at\":1515801301774,\"update_at\":1515801301774,\"edit_at\":0,\"delete_at\":0,\"is_pinned\":false,\"user_id\":\"mrjzniayh3fo9jut78uifrwoha\",\"channel_id\":\"etirmhnuctd7xgefxri6b6dg9a\",\"root_id\":\"\",\"parent_id\":\"\",\"original_id\":\"\",
		\"message\":\"mm vers xmpp\",
\"type\":\"\",\"props\":{},\"hashtags\":\"\",\"pending_post_id\":\"mrjzniayh3fo9jut78uifrwoha:1515801301701\"}", "sender_name":"SENDERNAME",
 "team_id":"trrjcgmydbd6zyfin15wred86h"}  module=mattermost

DEBU[2018-01-13T00:55:01+01:00] Receiving from matterclient 
&matterclient.Message{Raw:(*model.WebSocketEvent)(0xc42068ea80), Post:(*model.Post)(0xc4200a1440), Team:"alolise", Channel:"town-square", 
		Username:"SENDERNAME",
			 Text:"mm vers xmpp", 
Type:"", UserID:"mrjzniayh3fo9jut78uifrwoha"}  module=mattermost

DEBU[2018-01-13T00:55:01+01:00] Sending message from SENDERNAME on mattermost.alolise to gateway  module=mattermost

DEBU[2018-01-13T00:55:01+01:00] Message is config.Message{Text:"mm vers xmpp", Channel:"town-square", Username:"SENDERNAME", UserID:"mrjzniayh3fo9jut78uifrwoha", Avatar:"", Account:"mattermost.alolise", Event:"", Protocol:"", Gateway:"", Timestamp:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, ID:"6uho5jspibdnbmf5nq84dnwcnc", Extra:map[string][]interface {}{}}  module=mattermost

DEBU[2018-01-13T00:55:01+01:00] Sending config.Message{Text:"mm vers xmpp", Channel:"town-square", Username:"SENDERNAME", UserID:"mrjzniayh3fo9jut78uifrwoha", Avatar:"", Account:"mattermost.alolise", Event:"", Protocol:"", Gateway:"Alolise_gateway", Timestamp:time.Time{wall:0xbe8e7095745ad7b8, ext:30194051435, loc:(*time.Location)(0x14907c0)}, ID:"6uho5jspibdnbmf5nq84dnwcnc", Extra:map[string][]interface {}{}} from mattermost.alolise (town-square) to xmpp.alolise (espacepublic) 

DEBU[2018-01-13T00:55:01+01:00] Receiving config.Message{Text:"mm vers xmpp", Channel:"espacepublic", Username:" a ", UserID:"mrjzniayh3fo9jut78uifrwoha", Avatar:"", Account:"mattermost.alolise", Event:"", Protocol:"", Gateway:"Alolise_gateway", Timestamp:time.Time{wall:0xbe8e7095745ad7b8, ext:30194051435, loc:(*time.Location)(0x14907c0)}, ID:"", Extra:map[string][]interface {}{}}  module=xmpp

<message to="contact@alolise.org/2chjw4s2s9" type="groupchat" xml:lang="en" from="MYMUC@CONF.alolise.org/alolisebot"><body> a mm vers xmpp</body></message>

<iq to="SOMEONE@alolise.org/2chjw4s2s9" id="W8Dpl-2868" type="get" from="MYMUC@CONF.alolise.org/SENDERNAME"><query xmlns="http://jabber.org/protocol/disco#info"></query></iq>
```

### Any steps to reproduce the behavior?
Configure a MM and a XMPP MUC

#### (be sure to exclude or anonymize private data (tokens/passwords))
```dosini
[xmpp]

[xmpp.alolise]
Server="GOODXMPPSERV.alolise.org:5222"
Jid="GOODJID@alolise.org"
Password="GOOD PASS"
Muc="GOODMUCDOM.alolise.org"
Nick="GOOD NICK"
RemoteNickFormat=" a "
ShowJoinPart=true
SkipTLSVerify=true

[mattermost]

[mattermost.alolise]
Server="GOODSERV.alolise.org"
Team="alolise"
Login="GOOD USER"
Password="GOOD PASS"
NoTLS=false
SkipTLSVerify=false
NickFormatter="table"
NicksPerRow=4
PrefixMessagesWithNick=true
RemoteNickFormat="[{PROTOCOL}] <{NICK}> "
ShowJoinPart=true

[[gateway]]

name="Alolise_gateway"
enable=true

    [[gateway.inout]]
    account="xmpp.alolise"
    channel="espacepublic"

    [[gateway.inout]]
    account="mattermost.alolise"
    channel="town-square"
```

**Comment by 42wim** [comment]
You changed a lot in your debug so it's hard to follow

But this message
```
<message to="contact@alolise.org/2chjw4s2s9" type="groupchat" xml:lang="en" from="MYMUC@CONF.alolise.org/alolisebot"><body> a mm vers xmpp</body></message>
```

Should be (I think)
```
<message to="GOODJID@alolise.org/2chjw4s2s9" type="groupchat" xml:lang="en" from="espacepublic@GOODMUCDOM.alolise.org/alolisebot"><body> a mm vers xmpp</body></message>
```
Can you confirm that this is correct ?

**Comment by jadjay** [comment]
Yes that's correct
I had to change things to protect :relaxed:
But there is too much and i made mistakes. :no_mouth:
Sorry

**Comment by jadjay** [comment]
Yes that's correct
I had to change things to protect :relaxed:
But there is too much and i made mistakes. :no_mouth:
Sorry

**Comment by 42wim** [comment]
Ok, then I have no idea, it should work.
Could you try another xmpp server ? eg jabber.de to see if you've got the issue there too ?

**Comment by jadjay** [comment]
Well maybe it's me, and I don't know much of Xmpp but the way of this message seems odd to me : 
`<message to="GOODJID@alolise.org/2chjw4s2s9" type="groupchat" xml:lang="en" from="espacepublic@GOODMUCDOM.alolise.org/alolisebot">`
Look like it's the "espacepublic" which is talking to "GOODJID"...
The form : 
`<message to="espacepublic@GOODMUCDOM.alolise.org/alolisebot" type="groupchat" xml:lang="en" from="GOODJID@alolise.org/2chjw4s2s9">`
Looks better, because it's "GOODJID" which is talking IN "espacepublic" ...

I need to understand, it's not very important but I spent a night troubleshooting this bridge between mm and xmpp.

**Comment by jadjay** [comment]
Based on : https://tools.ietf.org/html/rfc6120#section-4.7
Seems legit ^^,

Same here : https://xmpp.org/extensions/xep-0045.html#message

**Comment by jadjay** [comment]
At least reading those specs, it seems okay to have this message FROM the MUC, but I think it's not good to SEND this to the MUC,
Know what I mean ?

**Comment by jadjay** [comment]
Idem on jabberfr.org, seems it comes from Matterbridge.

**Comment by jadjay** [comment]
Ok this is a bug from Xabber, you can close that !

Thanks a lot
